### PR TITLE
feature/v1-030/connection to API for My Courses

### PR DESCRIPTION
**Pull request by Julia-Maliutsina**

### CODE REVIEW

**Review by hauteballe — APPROVED** [review]
Great work :)

**Review by hauteballe — DISMISSED** [review]

File: frontend/src/types/course.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface Course {
+  title: string;
+  description: string;
+  technology: Array<string>;
+  requiredSkills: Array<string>;
+  duration: string;
+  testLink: string;
+  lessons: number;
```
Comment (SaXaPhonist) [inline frontend/src/types/course.ts:8]: also : 
```
_id?: string,
materials: Array<string>
 ```

File: frontend/src/pages/CoursesList/CoursesList.tsx [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+
+import { PageContainer, CourseButton, CourseActions } from './styled';
+
+const size = 'medium';
 
 const CoursesList: React.FC = () => (
   <AuthorizedLayout pageName="Courses List">
-    <div>Courses List should be here</div>
+    <PageContainer>
+      {INITIAL_COURSES.map((course, id) => (
```
Comment (SaXaPhonist) [inline frontend/src/pages/CoursesList/CoursesList.tsx:14]: don't use indx as key id, course has an unique field "_id"

File: frontend/src/api/myCourses/getMyCourses.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { Course } from 'types/course';
+
+const useGetMyCourses = (): UseQueryResult<Array<Course>, AxiosError> =>
+  useQuery('myCourses', async () => {
+    let myCoursesResponse: Array<Course>;
+    const apiClient = apiClientWrapper();
+    const userId = getUserIdCookie();
+    try {
+      const response = await apiClient.get(`${API.getMyCourses}/${userId}`);
+      myCoursesResponse = response.data;
```
Comment (SaXaPhonist) [inline frontend/src/api/myCourses/getMyCourses.ts:17]: you can use "const" here cause myCoursesResponse is never reassigned

File: frontend/src/pages/MyCourses/MyCourses.tsx [hunk 4]
```diff
@@ -1,67 +1,39 @@
...
-      ))}
-    </PageContainer>
-  </AuthorizedLayout>
-);
+const MyCourses: React.FC = () => {
+  const { data } = useGetMyCourses();
+  return (
+    <AuthorizedLayout pageName="My Courses">
+      <PageContainer>
+        {data?.map((course, id) => (
```
Comment (SaXaPhonist) [inline frontend/src/pages/MyCourses/MyCourses.tsx:15]: same

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: frontend/src/api/myCourses/getMyCourses.ts (hunk 3 above)
Comment (Julia-Maliutsina) [reply frontend/src/api/myCourses/getMyCourses.ts:17]: it is reassigned in line 17

**Review by hauteballe — APPROVED** [review]

File: frontend/src/api/myCourses/getMyCourses.ts (hunk 3 above)
Comment (SaXaPhonist) [reply frontend/src/api/myCourses/getMyCourses.ts:17]: yeah, but before it's "undefined"

**Review by SaXaPhonist — APPROVED** [review]

**Pull request merged**